### ISSUE TITLE
Allow DNS cache clean

### ISSUE THREAD

**Issue by iNPUTmice** [rtreffer/minidns#19]
Hi,

a user switching networks (eg from wifi to 3G) might get different DNS responses in different networks. For example a certain server reachable from the company wifi with an internal IP and from outside with an external IP. A user should be able to clear the DNS cache if it detects a network change.
I think this will be just a few lines of code and I'm going to send in a pull request but I want to run this by someone before I actually implementing this. Maybe I'm missing something in my thought process? Maybe this is already covered by another mechanism?

**Comment by Flowdalic** [comment]
> a user switching networks (eg from wifi to 3G) might get different DNS responses in different networks. For example a certain server reachable from the company wifi with an internal IP and from outside with an external IP

Those setups are fubar IMHO. DNS results should be globally unique.

Anyway, I think there is no need to change anything in MiniDNS, you can flush the cache already: Simply provide your own Cache which a `flush()` method when constructing the MiniDNS Client.

**Comment by rtreffer** [comment]
Well, I think the use case would be captive portals that return wrong records.... Not too uncommon (especially on mobile, try an open hotspot, notice a portal, move away -> dns cache poisoned)

I think it would be useful to provide a flush method for such a case (also note that resolvers may behave differently depending on IPv6 / IPv4 connectivity, up to the point that 6to4 requires a cleared DNS cache)
